packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -121,18 +120,7 @@ export function callTracedServerSideDataFetcher<F extends (...args: any[]) => Pr
     const currentScope = getCurrentHub().getScope();
     if (currentScope) {
       currentScope.setSpan(dataFetcherSpan);
-      currentScope.addEventProcessor(event =>
-        event.type !== 'transaction'
-          ? addRequestDataToEvent(event, req, {
-              include: {
-                // When the `transaction` option is set to true, it tries to extract a transaction name from the request
-                // object. We don't want this since we already have a high-quality transaction name with a parameterized
-                // route. Setting `transaction` to `true` will clobber that transaction name.
-                transaction: false,
```

Comment:
> It just took me 20 minutes to figure out why it is safe to remove the transaction: false logic here. I believe what confused me since we don't have the include.transaction defaulted to false in the requestdata integration. I think we should add that default just so it's more explicit.

`include.transaction` isn't a thing anymore, at least not for the integration. From the description of https://github.com/getsentry/sentry-javascript/pull/5703:

> - The options API for the new integration is inspired by, but different from, the options API for our Express request handler. 
>    - In the handler options, `transaction` can either be a boolean or a `TransactionNamingScheme`. In the integration, it can no longer be a boolean - events are going to have transactions, one way or another, so we shouldn't let people set it to `false`. Further, since it's now not about whether `transaction` is or isn't included, it's been moved out of `include` into its own `transactionNamingScheme` option. 

Regardless, we didn't actually need that setting in the first place, at least not for transaction events. Even if `include.transaction` is set to `true`, transaction events will never fall into the `if` because they always already have a `transaction` value: https://github.com/getsentry/sentry-javascript/blob/17e4cb8f460246ca22ef452f3957d222f4c94849/packages/node/src/requestdata.ts#L286

So that handles transaction events. As for error events, as of this PR it doesn't matter what `transaction` value `addRequestDataToTransaction` sets, because we're overwriting it in the integration.

> Also (for a lack of better place to comment) there are still some TODO comments in addRequestDataToEvent in requestdata.ts referencing Next.js. Can we remove those now?

Probably. I'll give them a look.

Reply:
Ah right. I now remember why I set `transaction: false` in the first place - it was because the errors originating from data fetchers had the unparameterized route in a tag - which now seems to be the case again and I'm not sure this is a good thing, since the `transaction` tag now mismatches the actual transaction (Do we even need to be exact here though? I guess it might be confusing to users):

Example - here the transaction tag is `GET /parameterized-page-with-getServerSideProps/two` while it should be `/parameterized-page-with-getServerSideProps/[pageNum]`

![Screen Shot 2022-09-20 at 10 43 13](https://user-images.githubusercontent.com/8118419/191211497-5bfaf560-f6a4-4cfb-97e9-dd4b8a9a9b3d.png)

![Screen Shot 2022-09-20 at 10 47 40](https://user-images.githubusercontent.com/8118419/191212580-845e970a-95ee-4bf3-841d-dddfc1c46fe0.png)

([event (after)](https://sentry.io/organizations/lforst-test-org/issues/3551336210/?project=6626560&query=is%3Aunresolved), [event (before)](https://sentry.io/organizations/lforst-test-org/issues/3551336210/events/8609685522c449148475e1e904675f9d/?project=6626560&query=is%3Aunresolved))

I guess both are sort of correct, I am just wondering which one is better. Do you have an opinion here?

Reply:
There also seem to be events where the tag is the request url. Not sure if this is related to this change though:

[event](https://sentry.io/organizations/lforst-test-org/issues/3551336210/events/fbf48ddde6774691a3c8e78dc76dc1f7/?project=6626560&query=is%3Aunresolved)

![Screen Shot 2022-09-20 at 10 56 24](https://user-images.githubusercontent.com/8118419/191214661-1d49c80c-51b3-4c2c-8fe7-cd9ee0bb2c5a.png)

Reply:
Huh. In my testing, the tag was always overwritten with the `transaction` value, even when I specifically set it to be something different in `beforeSend`. Are those events from this branch?

(Also, there's no reason we can't just fix the tag, too, just to be certain.)

Reply:
Yeah before is some event before this PR and after is after checking out the PR.

Are we sure we want to have the `GET` in the transaction tag though? Since there will be a mismatch between actual transaction and transaction tag.

Reply:
Yup, discovered that as well. I think I managed to drop a fixup commit somewhere in there, which both fixed how the transaction is retrieved and the check for whether or not to include the method for nextjs events. 

For nextjs, to make it match current behavior, the answer is [yes for API routes, no for page routes](https://github.com/getsentry/sentry-javascript/blob/17e4cb8f460246ca22ef452f3957d222f4c94849/packages/nextjs/src/utils/instrumentServer.ts#L268). I'm proposing that in v8 we should just standardize on always including the method. (We already include it in our [express transaction](https://github.com/getsentry/sentry-javascript/blob/master/packages/node/src/handlers.ts#L61-L74) names, for example.)

> there will be a mismatch between actual transaction and transaction tag.

From my testing, I believe the only way this can happen (in the UI, at least) is for an event to come in with a `transaction` tag but no top-level `transaction` value (like your ["before" event](https://sentry.io/organizations/lforst-test-org/issues/3551336210/events/8609685522c449148475e1e904675f9d/json/)). If an outgoing event has both a top-level `transaction` value and a `transaction` tag, and they don't match, ingest seems to overwrite the tag value with the top-level value. For example, this `beforeSend`

```js
beforeSend: (event) => {
  event.transaction = "aardvark";
  event.tags.transaction = "zebra";
  return event;
}
```

produced [an event](https://sentry.io/organizations/testorg-az/issues/3607054330/?project=5397699&query=is%3Aunresolved) with this JSON:

![image](https://user-images.githubusercontent.com/14812505/191365606-24169126-6546-4432-91e2-603f99834125.png)

Reply:
Gut feeling wise I like the current behavior (yes for API routes, no for page routes), because that way the pageload/pagenavigation transaction names and data-fetcher transactions line up. But let's discuss this in-depth when we get to it!

Reply:
Yeah, I'm not wedded to either outcome, I just think we should standardize it across SDKs if possible. The tricky part when it comes to something like Express is that it's not as clear-cut what's an API request and what's a page load request. But as you say, we can talk about it closer to the time.